+Base/QRconv.m

```diff
@@ -0,0 +1,384 @@
+function [v, V, options_fit, stages] = QRconv(img, options_guess, QR_parameters)
+    % QRconv finds QRs in an image via a convoluation algorithm.
+    % img is a NxM image.
+    % options_guess is a struct of the same format as options_fit: struct('ang', ..., 'calibration', ..., 'X_expected', ..., 'Y_expected', ...);
+    % v and V are arrays of column vectors (2xN) of the same size:
+    %  - v are positions of candidate QR codes in the coordinate system of the image img, while 
+    %  - V are the corresponding positons in QR space. 
+    % If a candidate does not pass the checksum, [NaN; NaN] is returned for V.
+    % Also returns transform from QR-space V to position-space v according to 
+    %   v = M * V + b
+    
+    % Default QR parameters.
+    if nargin < 3
+        QR_parameters = struct( 'r', .3, ...    % Circle radius,
+                                'l', 6.25, ...  % Arm length,
+                                'd', 40);       % QR spacing.
+    end
+    
+    % Step 0: prepare some helper variables
+    ang0 = options_guess.ang;
+    r = QR_parameters.r / options_guess.calibration;
+    l = QR_parameters.l / options_guess.calibration;
+    options_guess.d = QR_parameters.d;
+    
+    if isfield(options_guess, 'X_expected') && isfield(options_guess, 'Y_expected')
+        V_expected = [options_guess.X_expected; options_guess.Y_expected];
+    else
+        V_expected = [NaN; NaN];
```

Comment:
If there's only one of X or Y, can we include that guess and let the other be NaN?

Reply:
No. Guess only comes into play by checking distance between expected and measured, for validating situations where only one QR is detected. Having one coordinate be NaN would complicate the logic in an unnecessary way. If you have a use case for this, feel free to add this functionality.